### PR TITLE
Individual release view

### LINKED ISSUE #424: Enable pages to delete, hide, unhide a project/release/file
We need UI that will allow users to modify the things they register and upload to Warehouse.

### LINKED ISSUE #2807: Add release detail page
We need to add a release detail page and link to it from the releases page

The release detail page should display:
- the files for a release and a journal for the release
- the ability to delete a release

In `warehouse/templates/manage/project.html` we have two links to this page:

1. via the version number on the left of the table
2. via the 'edit' button in the dropdown

screenshot from 2018-01-17 06-57-12

These are currently commented out in the code.  We should enable them as part of this issue.

### LINKED ISSUE #2808: Add ability to delete release
The current UI allows users to do this via the releases page:

screenshot from 2018-01-17 06-57-12

screenshot from 2018-01-17 06-57-18

This code is currently commented out and should be enabled.

In addition, users should be able to delete a release via the release detail page (see  https://github.com/pypa/warehouse/issues/2807)

### PR DESCRIPTION
**Pull request by nlhkabu**
Fixes #424, fixes #2807, fixes #2808.

screenshot from 2018-02-02 18-08-08

screenshot from 2018-02-02 18-02-06

TODO:

- [x] Make link work when user clicks on 'collaborators' from this page (currently returning a 404)
- [x] Determine if the table is showing the most important information
- [x] Fix the layout on mobile/tablet
- [x] Hook up 'delete release' modal
- [x] Hook up 'delete file' modal.  Maybe make this smarter, rather than using a loop?
- [x] Make 'copy checksum' add a 'copied' message at the top of the page
- [ ] ~Add 'delete all files' shortcut (?)~

### CODE REVIEW

**Comment by nlhkabu** [comment]
hi @di @dstufft @ewdurbin - I need some feedback on what information is most useful to display in the table here.  It is all needed/used?  For example, what is the purpose of the comments?  Thanks.

**Comment by di** [comment]
@nlhkabu I think everything you have in the table should probably stay.

I'm not really sure how users are using the "Comment" field for a release, it's optional and there's no limit on length, so it's possible it could be quite long but in my experience it's mostly unused. Maybe we could remove the column and add some kind of a sub-row if the release has a comment?

> Add 'delete all files' shortcut (?)

Not sure this is necessary, a user will probably only want to delete a single file, or an entire release.

**Comment by di** [comment]
@nlhkabu I fixed the collaboration 404 and hooked up the 'delete release' modal. I'm seeing an issue where the modal doesn't dismiss when there is an error, could you take a look?

Still need to hook up individual file deletion.

**Comment by di** [comment]
@nlhkabu I hooked up individual file deletion as well, back to you!

**Comment by nlhkabu** [comment]
I've added the 'comment' field as a tooltip on the main table.  The content will truncate at 60 chars. 

screenshot from 2018-02-06 07-37-26

Responsive now working.  I've made the decision here to hide type and version on smaller screens (for lack of space):

screenshot from 2018-02-06 20-06-48

screenshot from 2018-02-06 20-07-02

screenshot from 2018-02-06 20-07-12

screenshot from 2018-02-06 20-07-23

screenshot from 2018-02-06 20-07-32

**Comment by nlhkabu** [comment]
fyi @di - I've again made the error messages more explicit, e.g.

screenshot from 2018-02-06 20-08-47

I will look into why the modals are staying open now :)

**Comment by di** [comment]
> fyi @di - I've again made the error messages more explicit

Oops, did I merge over this? Apologies if I did!

File: warehouse/templates/manage/release.html [hunk 1]
```diff
@@ -117,7 +117,7 @@ <h3>Delete Release</h3>
 
   <div id="delete-release-modal" class="modal">
     <div class="modal__content" role="dialog">
-      <form method="POST" class="modal__form">
+      <form method="POST" class="modal__form" action="{{ request.route_path('manage.project.release', project_name=project.name, version=release.version) }}">
```
Comment (di) [inline warehouse/templates/manage/release.html]: Aha! This makes sense. We can use `action="{{ request.current_route_path() }}"` here instead.

Comment (nlhkabu) [reply warehouse/templates/manage/release.html]: updated, thanks :)

**Comment by nlhkabu** [comment]
Checksums modal:

screenshot from 2018-02-09 07-32-49

screenshot from 2018-02-09 07-33-14

screenshot from 2018-02-09 07-33-26

**Comment by nlhkabu** [comment]
Hi @di - this should be ready for final review and merge :)

**Pull request merged**